sdk/storage/azure-storage-common/src/file_io.cpp

```diff
@@ -72,12 +79,26 @@ namespace Azure { namespace Storage { namespace _internal {
 
   FileWriter::FileWriter(const std::string& filename)
   {
+    int sizeNeeded
+        = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, filename.data(), -1, nullptr, 0);
+    if (sizeNeeded == 0)
+    {
+      throw std::runtime_error("invalid filename");
+    }
+    std::wstring filenameW(sizeNeeded, L'\0');
+    if (MultiByteToWideChar(
+            CP_UTF8, MB_ERR_INVALID_CHARS, filename.data(), -1, &filenameW[0], sizeNeeded)
+        == 0)
+    {
+      throw std::runtime_error("invalid filename");
+    }
+
```

Comment:
Ditto my above comments. (I'd extract this common sequence into a function that widens strings for filenames.)